components/hab/src/error.rs

```diff
@@ -127,6 +128,13 @@ impl fmt::Display for Error {
             Error::HabitatCore(ref e) => format!("{}", e),
             Error::HandlebarsRenderError(ref e) => format!("{}", e),
             Error::IO(ref err) => format!("{}", err),
+            Error::JobGroupPromote(ref id, ref c) => {
+                format!(
+                    "Failed to promote job group ID {} to channel {} because the job group is not complete.",
```

Comment:
Hmm, the user already has to know what a job group and a channel is to use this CLI command.  We've decided against using the term "project" in any user facing text, to avoid confusion with packages. There are 4 possible reasons that a project might not've succeeded, but listing them all out here feels a bit overly verbose.